### DISCUSSION TITLE
Log Rotation

### DISCUSSION THREAD

**Discussion by clarkholdham** [fluent/fluentd#4164] [Q&A]
I'm having some trouble interpreting the [Log Rotation Setting](https://docs.fluentd.org/deployment/logging#log-rotate-size-bytes) documentation. Specifically the rotate_age option. The docs specify this can be an integer or string value. My understanding is if this field is an integer value, the field indicates "how many logs to keep before removing the oldest" but, when this field is a string, the field indicates "when to rotate a log file" (ex daily, weekly, monthly). These don't seem like they are the same function to me?

Essentially what I'm trying to achieve is a daily log rotation, keeping only the last 7 logs (1 week). If I were to set the rotate_age to "daily", I'm not sure how I would specify the "age" (number) of files to keep. Is this configuration possible with some combination of these settings?

**Comment by kenhys** [comment]
In such a situation, it may be better to delegate rotation to  /etc/logrotate.d/ configuration.
